README.md

```diff
@@ -42,7 +42,7 @@ The A2A Java SDK provides a Java server implementation of the [Agent2Agent (A2A)
 ```xml
 <dependency>
     <groupId>io.a2a.sdk</groupId>
-    <artifactId>a2a-java-sdk-core</artifactId>
+    <artifactId>a2a-java-sdk-client</artifactId>
```

Comment:
This should be `a2a-java-sdk-spec`

Reply:
In the README, the "Add the A2A Java SDK Core Maven dependency to your project" should be reworded to "Add the A2A Java SDK Spec Maven dependency to your project"

Reply:
Hum not sure about this one tbh.

Now that we splitted client and server modules, there should be two different imports:
* One for `a2a-java-sdk-client` if you want to use the client part of the SDK
* One for `a2a-java-sdk-server-common` if you want to use the server part of the SDK

Reply:
This part of the README goes through the steps for creating an A2A server agent. In this first step, the `a2a-java-sdk-spec` dependency needs to be added in order to be able to create the `AgentCard` producer and `AgentExecutor` producer. A later step shows how to add the `a2a-java-sdk-server-quarkus` or `a2a-java-sdk-server-jakarta` dependency to expose the A2A server endpoint. The client dependency shouldn't be needed for creating a server agent.

Reply:
I guess the dependency on `a2a-java-sdk-server-quarkus` or `a2a-java-sdk-server-jakarta` would also bring in the dependency on `a2a-java-sdk-spec`. So I guess the question is if we want to be explicit about it or not.

Let's see what @kabir thinks.